### PR TITLE
rustdoc-json: add type/trait alias tests

### PR DESCRIPTION
**Pull request by its-the-shrimp**
Not sure if this tests everything there is to test in them though.

Updates #81359

### CODE REVIEW

File: tests/rustdoc-json/trait_alias.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// @has "$.index[*][?(@.name=='StrLike')].inner.trait_alias"
+// @is "$.index[*][?(@.name=='StrLike')].span.filename" $FILE
+pub trait StrLike = AsRef<str>;
+
+// @is "$.index[*][?(@.name=='f')].inner.function.decl.output.impl_trait[0].trait_bound.trait.id" $StrLike
+pub fn f() -> impl StrLike {
+    "heya"
+}
+
+// @!is "$.index[*][?(@.name=='g')].inner.function.decl.output.impl_trait[0].trait_bound.trait.id" $StrLike
```
Comment (compiler-errors) [inline tests/rustdoc-json/trait_alias.rs:15]: Why does this test pass?? the user didn't write `StrLike` here?

Comment (its-the-shrimp) [reply tests/rustdoc-json/trait_alias.rs:15]: Yes, I don't know if it even makes sense to test this from the perspective of the implementation of rustc/rustdoc, but who knows, maybe someday it'll accidentally start conflating trait aliases and their equivalent trait, then this test case will heroically prevent a disaster

Comment (compiler-errors) [reply tests/rustdoc-json/trait_alias.rs:15]: Oh - I am confused. I missed the `!`.

Comment (compiler-errors) [reply tests/rustdoc-json/trait_alias.rs:15]: That is to say, I thought this was testing *for* the trait alias, not testing for the *lack* of it.

**Comment by aDotInTheVoid** [comment]
Thanks! As a potential followup, it'd be nice to also test for these when re-exporting, as that distinction seems more susceptible to being lost there. But that isn't necessary to make this worth merging.

@bors r+ rollup

**Pull request merged**